common/bytes.go

```diff
@@ -77,13 +77,25 @@ func Bytes2Hex(d []byte) string {
 
 // Hex2Bytes returns the bytes represented by the hexadecimal string str.
 func Hex2Bytes(str string) []byte {
-	h, _ := hex.DecodeString(str)
+	if !isHex(str) {
+		return nil
+	}
+	h, err := hex.DecodeString(str)
```

Comment:
The only part about this PR I think should be kept is that we return a nil slice if `DecodeString` fails.  The current behavior is:
> If the input is malformed, DecodeString returns the bytes decoded before the error.

Reply:
On the other hand, it's explicitly expected that these functions take valid hex strings.  So probably we can just close the PR entirely.